### PR TITLE
Show usage and exit(0) if no args provided on cli-tools

### PR DESCRIPTION
**Pull request by fambon**

### CODE REVIEW

File: iotlabcli/parser/common.py [hunk 1]
```diff
@@ -128,6 +128,10 @@ def catch_missing_auth_cli():
 def main_cli(function, parser, args=None):  # flake8: noqa
     """ Main command-line execution. """
     args = args or sys.argv[1:]
+    if len(args) == 0:
```
Comment (aabadie) [inline iotlabcli/parser/common.py]: Maybe a test could be added somewhere [around there](https://github.com/iot-lab/cli-tools/blob/master/iotlabcli/tests/common_parser_test.py#L37)

File: iotlabcli/integration/test_cli_tools_help.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import pytest
```
Comment (aabadie) [inline iotlabcli/integration/test_cli_tools_help.py]: Don't take it bad but I think that is already covered by https://github.com/iot-lab/cli-tools/blob/master/tox.ini#L28 in the CI.
It really seems to me that you are duplicating what already exists.
The output of the `--help` option can (should) be tested in the parser test module.

**Comment by aabadie** [comment]
General comment : the PR title doesn't fully reflects the changes here.
Why creating another integration directory when [good tests and CI already exist](https://ci.inria.fr/iot-lab/job/cli-tools-tox/) ?

File: iotlabcli/integration/test_cli_tools_help.py (hunk 2 above)
Comment (fambon) [inline iotlabcli/integration/test_cli_tools_help.py]: The test you mention in tox.ini does not cover the use-cases that the pytest test does.  I have seen this bash for loop and decided to provide a systematic, "as a user would do" test case, of all cli tools menus/sub-menus, thus the pytest.

File: iotlabcli/parser/common.py (hunk 1 above)
Comment (fambon) [inline iotlabcli/parser/common.py]: upon your request, I looked "around there" and proposed instead to add the test here: https://github.com/iot-lab/cli-tools/blob/show_usage_if_no_args/iotlabcli/integration/test_cli_tools_help.py#L20

whiche slightly extended into the file as it is now.

**Comment by fambon** [comment]
About the PR title: the pull request morphed into a bigger "test" content following your suggestion of adding a test for the proposed feature.  However, the PR is, as far as I can tell, about "Showing usage and exit(0) if no args provided on cli-tools".  And now comes with a test for this feature (plus a few others).

File: iotlabcli/integration/test_cli_tools_help.py (hunk 2 above)
Comment (aabadie) [inline iotlabcli/integration/test_cli_tools_help.py]: I know. But the tox.ini verifies that the script can be called and that's it. The internal behavior : checking that the options are taken into account correctly is done in the parser tests.
We don't need to write python code that tests things already tested somewhere else.

Comment (fambon) [inline iotlabcli/integration/test_cli_tools_help.py]: Talking about the parser tests, these are "dev unit-tests", hitting on internal code, not "integration tests", hitting on user-visible features.  Sure they both "sort of" test the same thing ; I see them as complimentary.  Are we bickering for 30 sloc ? No big deal, right ?

I might be biased, but I find the pytest-based test to be neat, simple, all in a single place, for all tools ; and it shows all available cli interface tools and commands in a snap.

**Pull request closed without merging**